packages/frontend/editor-ui/src/features/dataStore/composables/useDataStoreGridBase.ts

```diff
@@ -0,0 +1,476 @@
+			editable: false,
+			suppressMovable: true,
+			lockPinned: true,
+			lockPosition: 'right',
+			headerComponentParams: {
+				allowMenuActions: false,
+			},
+		};
+		return [
+			// Always add the ID column, it's not returned by the back-end but all data stores have it
+			// We use it as a placeholder for new datastores
+			createColumnDef(
+				{
+					index: 0,
+					id: DEFAULT_ID_COLUMN_NAME,
+					name: DEFAULT_ID_COLUMN_NAME,
+					type: 'string',
+				},
+				{
+					editable: false,
+					sortable: false,
+					suppressMovable: true,
+					headerComponent: null,
+					lockPosition: true,
+					minWidth: DATA_STORE_ID_COLUMN_WIDTH,
+					maxWidth: DATA_STORE_ID_COLUMN_WIDTH,
+					resizable: false,
+					cellClass: (params) =>
+						params.data?.id === ADD_ROW_ROW_ID ? 'add-row-cell' : 'id-column',
+					cellRendererSelector: (params: ICellRendererParams) => {
+						if (params.value === ADD_ROW_ROW_ID) {
+							return {
+								component: AddRowButton,
+								params: { onClick: onAddRowClick },
+							};
+						}
+						return undefined;
+					},
+				},
+			),
+			// Append other columns
+			...orderBy(dataStoreColumns, 'index').map((col) => createColumnDef(col)),
+			createColumnDef(
+				{
+					index: dataStoreColumns.length + 1,
+					id: 'createdAt',
+					name: 'createdAt',
+					type: 'date',
+				},
+				systemDateColumnOptions,
+			),
+			createColumnDef(
+				{
+					index: dataStoreColumns.length + 2,
+					id: 'updatedAt',
+					name: 'updatedAt',
+					type: 'date',
+				},
+				systemDateColumnOptions,
+			),
+			createColumnDef(
+				{
+					index: dataStoreColumns.length + 3,
+					id: 'add-column',
+					name: 'Add Column',
+					type: 'string',
+				},
+				{
+					editable: false,
+					suppressMovable: true,
+					lockPinned: true,
+					lockPosition: 'right',
+					resizable: false,
+					headerComponent: AddColumnButton,
+					headerComponentParams: { onAddColumn },
+				},
+			),
+		];
+	};
+
+	const loadColumns = (dataStoreColumns: DataStoreColumn[]) => {
+		colDefs.value = getColumnDefinitions(dataStoreColumns);
+		setGridData({ colDefs: colDefs.value });
+	};
+
+	const deleteColumn = (columnId: string) => {
+		colDefs.value = colDefs.value.filter((col) => col.colId !== columnId);
+		setGridData({ colDefs: colDefs.value });
+	};
+
+	const insertColumn = (column: ColDef, index: number) => {
+		colDefs.value.splice(index, 0, column);
+		setGridData({ colDefs: colDefs.value });
+	};
+
+	const addColumn = (column: DataStoreColumn) => {
+		colDefs.value = [
+			...colDefs.value.slice(0, -1),
+			createColumnDef(column),
+			...colDefs.value.slice(-1),
+		];
+		setGridData({ colDefs: colDefs.value });
+	};
+
+	const moveColumn = (oldIndex: number, newIndex: number) => {
+		const fromIndex = oldIndex - 1; // exclude ID column
+		const columnToBeMoved = colDefs.value[fromIndex];
+		if (!columnToBeMoved) {
+			return;
+		}
+		const toIndex = newIndex; // exclude selection + ID columns used by AG Grid indices
```

Comment:
This seemed a bit too involved to me so I tried simpler version without `reorderItem`:
```ts
const moveColumn = (oldIndex: number, newIndex: number) => {
	const fromIndex = oldIndex - 1; // exclude ID column
	const columnToBeMoved = colDefs.value[fromIndex];
	if (!columnToBeMoved) {
		return;
	}
	const middleColumns = [...colDefs.value.slice(1, -1)];
	const [movedColumn] = middleColumns.splice(fromIndex, 1);
	middleColumns.splice(newIndex, 0, movedColumn);
	colDefs.value = [colDefs.value[0], ...middleColumns, colDefs.value[colDefs.value.length - 1]];
};
```
Not sure if I missed something so take a look if that makes sense.

Reply:
Yeah, I think this is missing the part that if we're moving the item forward, once we remove it (the splice part), the toIndex will not be accurate anymore. I prefer the helper function as we're using it in one more place and also it's easier to unit test and make sure it works as expected